scripts/cabal-format.sh

```diff
@@ -2,6 +2,23 @@
 
 set -euo pipefail
 
-for f in $(git ls-files -- '*.cabal'); do cabal format $f; done
+if [[ "${1:-format}" != "format" && "${1:-format}" != "check" ]]; then
+  echo "Error: Invalid mode. Allowed values are 'format' or 'check'." >&2
+  exit 1
+fi
 
-git diff --exit-code
+mode="${1:-format}"
+
+git ls-files -- '*.cabal' | while IFS= read -r f; do
```

Comment:
It would be good to also add `cabal.project` to this list.

Reply:
done in a separate commit, great idea, thanks.